### PR TITLE
Wallet Labels Export Format

### PR DESCRIPTION
**Pull request by craigraw**
This was first proposed [on the mailing list](https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2022-August/020887.html) on Aug 24 as a common format for the export and import of wallet labels.

Following discussion there, on this [bitcointalk post](https://bitcointalk.org/index.php?topic=5411159.0) and via private channels with other wallet developers, the original proposal was simplified, and changed to use JSON instead of CSV. The revised proposal has been well received, and I believe is ready for consideration as a BIP.

Link: https://github.com/craigraw/bips/blob/master/bip-wallet-labels.mediawiki

### CODE REVIEW

**Comment by abdul-hannan1** [comment]
I have address bsc

**Review by moneymanolis — APPROVED** [review]

**Comment by moneymanolis** [comment]
What's blocking the merge of this?

**Review by k9ert — APPROVED** [review]

File: bip-wallet-labels.mediawiki [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+| Transaction id in hexadecimal format
+| <tt>f91d0a8a78462bc59398f2c5d7a84fcff491c26ba54c4833478b202796c8aafd</tt>
+|-
+| <tt>addr</tt>
+| Address in base58 or bech32 format
+| <tt>bc1q34aq5drpuwy3wgl9lhup9892qp6svr8ldzyy7c</tt>
+|-
+| <tt>pubkey</tt>
+| 32 or 33 byte public key in hexadecimal format
+| <tt>0283409659355b6d1cc3c32decd5d561abaac86c37a353b52895a5e6c196d6f448</tt>
```
Comment (luke-jr) [inline bip-wallet-labels.mediawiki:82]: What about 65-byte pubkeys?

**Comment by luke-jr** [comment]
Assigned BIP 329

File: bip-wallet-labels.mediawiki (hunk 1 above)
Comment (craigraw) [reply bip-wallet-labels.mediawiki]: Good point, added.

**Comment by dipunm** [comment]
I get that this is closed, but am new to commenting on bips so I'll add my comment here.

> An importing wallet may ignore records it does not store, and truncate labels if necessary.

I think adding truncation as a standard is dangerous. Context can be lost, especially if context was at the end. Also different apps may have further restrictions such as character limitations (no emoticon support or Chinese character support)

Is it not more appropriate to warn that care should be taken if truncating or modifying the label and/or a warning should be given if any modifications will be made. UI is up to the app.

Something like that?

File: bip-0329.mediawiki [hunk 2]
```diff
@@ -1,10 +1,10 @@
 <pre>
-  BIP: wallet-labels
+  BIP: 329
   Layer: Applications
   Title: Wallet Labels Export Format
   Author: Craig Raw <craig@sparrowwallet.com>
   Comments-Summary: No comments yet.
-  Comments-URI: https://github.com/bitcoin/bips/wiki/Comments:BIP-wallet-labels
+  Comments-URI: https://github.com/bitcoin/bips/wiki/Comments:BIP-0326
```
Comment (luke-jr) [inline bip-0329.mediawiki]: Wrong number here...

**Pull request merged**